resources/resource.py

```diff
@@ -0,0 +1,89 @@
+#  Aristotle Learning Platform: Luscinia Enterprises Assn.
+#  Copyright (C) 2020
+#      1261612 B.C. LTD.
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero General Public License as
+#  published by the Free Software Foundation, either version 3 of the
+#  License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Affero General Public License for more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from flask import Response, request
+from flask_restful import Resource as Restful_Resource
+from mongoengine import *
+from dateutil.parser import *
+import uuid
+
+from models.Resource import Resource
+
+
+class ResourcesApi(Restful_Resource):
+    def get(self):
+        # Retrieves all resources
+        resources = Resource.objects.to_json()
+        return Response(resources, mimetype="application/json", status=200)
+
+    def post(self):
+        # Creates new resource and returns the id if success
+        body = request.get_json()
+
+        # manually parse dates and uuid
+        body["copyrightInfo"]["publication"] = isoparse(body["copyrightInfo"]["publication"]["$date"])
+        body["copyrightInfo"]["update"] = isoparse(body["copyrightInfo"]["update"]["$date"])
+        body["uuid"] = uuid.UUID(body["uuid"]["$uuid"])
```

Comment:
What format is needed to not require parsing?
Java does not return Unix timestamps, it returns format yyyy-MM-dd.

Reply:
I tried them all and I couldn't figure out any format that did not require manual parsing. Given that this part of the code, creating and changing resources, isn't going to be used in the final api, I think manual parsing of Java's ISODate format will have to do for now.

Reply:
There are 3 ways dates are stored right now as i see it

1. Java - yyyy-MM-dd
2. MongoDB - Unix Timestamp?
3. Python - ISODate

What I'm trying to get at is that since this is being used as template code for database interactions, we can simplify by returning raw values (or at least values that python is capable of understanding, manipulating and writing back to db with the least number of steps.

Reply:
Ok that makes sense. The problem is that I can't figure out how to parse the json dates without using manual parsing. 
Right now in the db, I see dates are stored as ISO.

"publication": {"$date": "2015-01-11T00:00:00.000Z"}

Python can't parse this date itself and raises validationerror. So after manually parsing that date, mongoengine saves the new resource object with a date in unix timestamp:
"publication": {  "$date": 1597622400000  }

However, if I give python that format to parse without doing any manual parsing, Mongoengine still fails to parse that date, raising ValidationError, cannot parse date {'$date': 1597622400000}.

Mongoengine saves the date in that unix timestamp format and can't even parse it itself???

The only format that I have found that doesn't require manual parsing is 
"publication": "2015-01-11T00:00:00.000Z"

but mongoengine doesn't save dates in that format.

Overall, I have absolutely no idea how to parse the date without manual parsing.

Reply:
Ok, lets stick with what you have since the only dates exist in copyrightInfo which will not be handled by the api.
Approving Review.